syntax.php

```diff
@@ -3,135 +3,89 @@
  * @license    GPL 2 (http://www.gnu.org/licenses/gpl.html)
  * @author     Andreas Gohr <andi@splitbrain.org>
  */
-// must be run within Dokuwiki
-if(!defined('DOKU_INC')) die();
 
-if(!defined('DOKU_PLUGIN')) define('DOKU_PLUGIN',DOKU_INC.'lib/plugins/');
-require_once(DOKU_PLUGIN.'syntax.php');
+class syntax_plugin_xref extends DokuWiki_Syntax_Plugin
+{
 
-class syntax_plugin_xref extends DokuWiki_Syntax_Plugin {
+    protected $dir = '';
+    protected $web = '';
 
-    var $dir = '';
-    var $web = '';
-
-    function syntax_plugin_xref(){
-        $this->dir = rtrim($this->getConf('dir'),'/');
-        $this->web = rtrim($this->getConf('web'),'/');
+    public function __construct()
+    {
+        $this->dir = rtrim($this->getConf('dir'), '/');
+        $this->web = rtrim($this->getConf('web'), '/');
     }
 
-    /**
-     * What kind of syntax are we?
-     */
-    function getType(){
+    /** @inheritdoc */
+    public function getType()
+    {
         return 'substition';
     }
 
-    /**
-     * What about paragraphs?
-     */
-    function getPType(){
+    /** @inheritdoc */
+    public function getPType()
+    {
         return 'normal';
     }
 
-    /**
-     * Where to sort in?
-     */
-    function getSort(){
+    /** @inheritdoc */
+    public function getSort()
+    {
         return 150;
     }
 
-
-    /**
-     * Connect pattern to lexer
-     */
-    function connectTo($mode) {
-        $this->Lexer->addSpecialPattern('\[\[xref>.+?\]\]',$mode,'plugin_xref');
+    /** @inheritdoc */
+    public function connectTo($mode)
+    {
+        $this->Lexer->addSpecialPattern('\[\[xref>.+?\]\]', $mode, 'plugin_xref');
     }
 
+    /** @inheritdoc */
+    public function handle($match, $state, $pos, Doku_Handler $handler)
+    {
+        $match = trim(substr($match, 7, -2));
 
-    /**
-     * Handle the match
-     */
-    function handle($match, $state, $pos, Doku_Handler $handler){
-        $match = trim(substr($match,7,-2));
-
-        list($link,$name) = explode('|',$match,2);
-        list($link,$anchor) = explode('#',$link,2);
-        if(!$name) $name = $link;
-        if($anchor) $anchor = "#".$anchor;
+        list($reference, $name) = explode('|', $match, 2);
+        list($reference, $anchor) = explode('#', $reference, 2);
+        if (!$name) $name = $reference;
+        if ($anchor) $reference = "#" . $anchor;
```

Comment:
The intention is to overwrite the reference and not to append the anchor?
I thought the parts together are more unique.
Hmm, but now I’m trying a bit, I do not recognize it at the grok site.

Reply:
No, this here just just breaks the syntax apart. It basically removes the anchor, uses the remainder as the link name and then re-adds the anchor to pass it to the Grok class later. I guess it would actually be simpler to pass on `$match` to Grok.

Reply:
Ah no match also contains the `|title` part so this is correct

Reply:
BTW these kind of anchors aren't used in Grok at all, but we did use them to reference things in a specific file in phpxref.

So `[[xref>/inc/foo.php#something]]` means find a function, constant or variable named `something` in `/inc/foo.php`. The Heuristics class takes that input and creates the proper search setup for that.

Reply:
check, for backward compatibility. 
I observe the 'Definition' field of Grok is case-sensitive. So more anchors are not found. This is ok, because it is also ok to update these usage of [[xref>]].

Reply:
resolved for me.